src/v/raft/configuration.cc

```diff
@@ -176,18 +183,38 @@ void group_configuration::remove(const std::vector<model::node_id>& ids) {
 }
 
 void group_configuration::replace(std::vector<model::broker> brokers) {
-    vassert(
-      !_old, "can not remove broker from joint configuration - {}", *this);
+    vassert(!_old, "can not replace joint configuration - {}", *this);
 
     // add missing brokers, brokers have to contain both the new and old nodes
+
+    if (brokers == _brokers) {
+        bool has_all = std::all_of(
+          brokers.begin(), brokers.end(), [this](model::broker& b) {
+              return _current.contains(b.id());
+          });
+        // configurations are identical, do nothing
```

Comment:
this comment seems to be satisfied by `brokers == _brokers` i think a comment about why the `std::all_of` is necessary would be helpful to understanding this condition.

Reply:
I've added a comment in the code explaining why we have to check both condition. We have list of brokers plus two lists of ids i.e. voters and learners. The `brokers == _brokers` check only the brokers list.